src/utils/message.ts

```diff
@@ -19,13 +20,49 @@ export const messageHandlerDict: Record<
   (message: Message, connection: DataConnection) => void
 > = {
   [MessageType.UPDATE_PLAYER_NAME]: handlePlayerNameMessage,
+  [MessageType.UNAVAILABLE_PLAYER_NAME]: handleUnavaiablePlayerNameMessage,
   [MessageType.UPDATE_PLAYER_LIST]: handlePlayerListMessage,
   [MessageType.CHAT_MESSAGE]: handleChatMessage,
 };
 
-function handlePlayerNameMessage(message: Message, connection: DataConnection) {
+function disconnectFromHost(connection: DataConnection) {
```

Comment:
I think it makes more sense to disconnect from the host side, but that might require some error handling improvements first. Can you add a `// TODO` here?

Reply:
UPDATE: With #16, we can just do a `connection.close()` from the host side and the other side would exit room automatically when disconnected.